pallets/qa-tools/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+    };
+    use frame_support::dispatch::DispatchErrorWithPostInfo;
+    use frame_support::traits::Get;
+    use frame_support::{dispatch::PostDispatchInfo, pallet_prelude::*};
+    use frame_system::pallet_prelude::*;
+    use order_book::{MomentOf, OrderBookId};
+    use sp_std::prelude::*;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + order_book::Config + trading_pair::Config {
+        type WeightInfo: WeightInfo;
+        type OrderBookOrderLifespan: Get<MomentOf<Self>>;
+        type AssetInfoProvider: AssetInfoProvider<
+            Self::AssetId,
+            Self::AccountId,
+            AssetSymbol,
+            AssetName,
+            BalancePrecision,
+            ContentSource,
+            Description,
+        >;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Order book related errors
+        OrderBook(pallets::order_book_tools::Error),
+    }
+
+    #[derive(
+        Encode,
+        Decode,
+        Eq,
+        PartialEq,
+        Copy,
+        Clone,
+        PartialOrd,
+        Ord,
+        RuntimeDebug,
+        Hash,
+        scale_info::TypeInfo,
+        MaxEncodedLen,
+    )]
+    pub struct OrderBookFillSettings {
+        /// Best (highest) price for placed buy orders
+        pub best_bid_price: order_book::types::OrderPrice,
+        /// Best (lowest) price for placed sell orders
+        pub best_ask_price: order_book::types::OrderPrice,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Create multiple order books with default parameters (if do not exist yet).
+        ///
+        /// Parameters:
+        /// - `origin`: caller, should be account because error messages for unsigned txs are unclear,
+        /// - `order_book_ids`: ids of the created order books; trading pairs are created
+        /// if necessary,
+        #[pallet::call_index(0)]
+        #[pallet::weight(<T as Config>::WeightInfo::order_book_create_empty_batch())]
+        pub fn order_book_create_empty_batch(
+            origin: OriginFor<T>,
+            order_book_ids: Vec<OrderBookId<T::AssetId, T::DEXId>>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            // replace with more convenient `with_pays_fee` when/if available
+            // https://github.com/paritytech/substrate/pull/14470
+            pallets::order_book_tools::create_multiple_empty_unchecked::<T>(&who, order_book_ids)
+                .map_err(|e| DispatchErrorWithPostInfo {
+                post_info: PostDispatchInfo {
+                    actual_weight: None,
+                    pays_fee: Pays::No,
+                },
+                error: e,
+            })?;
+
+            // Extrinsic is only for testing, so we return all fees
+            // for simplicity.
+            Ok(PostDispatchInfo {
+                actual_weight: None,
+                pays_fee: Pays::No,
```

Comment:
I don't think it's good idea to have such extrinsics on stage or even test environment. We can mint as many tokens on test environments as we want for testing, but some bad users can try to break our testnets with this extrinsics. Another way to avoid such problem is to make this extrinsics callable only by root or trusted account

Reply:
Since we try to "record" the development process with comments on issues (also why not PRs), @ra9mls, can you clarify please?)

Reply:
I think we can add storage with whitelisted accounts to this pallet and add new accounts to whitelist via root exrinsic

Reply:
It's good to have weights for this extrinsics, but it's not so big problem if we will have restricted access to this pallet

Reply:
> I think we can add storage with whitelisted accounts to this pallet and add new accounts to whitelist via root exrinsic

Seems good! Just want to clarify, @ra9mls, is this approach convenient from QA perspective?

Reply:
Got the answer in DM, it's ok.